gelic-idealab/komodo-unity
# Capture Button Spectator & Capture Started Event

# Capture Button Spectator & Capture Started Event


## Redesigned UI for both VR and desktop modes
The first feature in this pull request is the Capture Button Spectator. This feature redesigned the capture button in both the VR mode and the desktop mode; the capture button is now located at the Instructor Menu, which is exclusive on the desktop mode. For VR users, the capture button is currently disabled and not supported.

### Type of change


### How Has This Been Tested?

1. Enter play mode.
2. Click Settings tab.
3. Click Instructor Menu.
4. Try out the Capture button.
5. Look in the Console > Info messages for `[SocketSim] Emit(start_recording, 0)` and `[SocketSim] Emit(end_recording, 0)`

To check the UI for VR mode,
1. Connect to an Oculus VR device
2. Look through the menu.
3. The Instructor Menu button shouldn't be in the menu. 

_REPLACE ME: any relevant details for your test configuration_



## ~~captureStarted event~~
~~The client side is now able to receive a captureStarted event sent by the server; once client receives the event from the server, it will disable the capture button for all other users in the room.~~ 
[This feature is now **removed** from the current PR. The source code is now kept at the **legacy/capture-started-event** branch.

### How Has This Been Tested?

~~1. Enter play mode through Unity Editor.~~
~~2. Look for **VisibleManagers** in the Hierarchy~~
~~3. Click the **NetworkManager** in **VisibleManagers**~~
~~4. In the Inspector of **NetworkManager**, right click on **Socket IO Editor Simulator (script)** and select **DisableCaptureButton**.~~
~~5. The capture button should be disable.~~
_____

# Test Configuration

* Oculus II
* Unity Editor
- [x] PC VR (**Link** / AirLink)

# Checklist:


Modified from this article:
Phillip Johnston, “A GitHub Pull Request Template for Your Projects - Embedded Artistry,” Embedded Artistry, Aug. 04, 2017. https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ (accessed Jul. 22, 2021).
